### PR TITLE
fix: css modules with hmr

### LINKED ISSUE #639: Suspense doesn't work in layout with css modules (DEV-only)
We have a tricky case in an example.

https://github.com/dai-shi/waku/blob/3b87654f39fc12592b1238e89e3423f105f41840/examples/09_cssmodules/src/components/Layout.tsx#L6-L20

It should show "Pending..." and then "Hello from server!".
However, it doesn't work as expected.

It seems like a regression in #446.
This is a dev-only issue.
@Aslemammad Can you investigate it please?

Here is a screencast that works before #446 commit.

https://github.com/dai-shi/waku/assets/490574/3e082901-5662-4360-84ef-085ae81c28ff

### PR DESCRIPTION
**Pull request by dai-shi**
close #639

### CODE REVIEW

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts [hunk 1]
```diff
@@ -194,38 +194,26 @@ async function generateInitialScripts(
...
-        injectedBlockingViteClient = true;
-      }
+      // CSS modules do not support result.source (empty) since ssr-transforming them gives the css keys
+      // and client-transforming them gives the script tag for injecting them.
       scripts.push({
         tag: 'script',
-        // tried render blocking this script tag by data url imports but it gives `/@vite/client: Invalid relative url or base scheme isn't hierarchical.` which could not find a way to fix.
-        attrs: { type: 'module', 'waku-module-id': result.id },
+        // async: true to wait vite/client to be loaded
+        attrs: { type: 'module', async: true, 'waku-module-id': result.id },
```
Comment (Aslemammad) [inline packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:205]: I guess this would cause FOUC! Since css will potentially load later.

Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:205]: yeah, but better than waiting everything.

Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:205]: so, if there are a better solution, it's better.

**Review by Aslemammad — APPROVED** [review]

File: packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts (hunk 1 above)
Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:205]: On another thought, I'm not sure if it really causes FOUC. Let's reconsider if it happens.

Comment (dai-shi) [reply packages/waku/src/lib/plugins/vite-plugin-rsc-hmr.ts:205]: Yeah, it causes FOUC. I found a better fix: #659 
btw `async: true` makes the execution earlier.

**Pull request merged**